packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts

```diff
@@ -142,7 +149,8 @@ export class LambdaInvoke extends sfn.TaskStateBase {
    * @internal
    */
   protected _renderTask(): any {
-    if (this.props.payloadResponseOnly) {
+    if (this.props.payloadResponseOnly &&
+      (this.props.payload?.value === undefined || typeof(this.props.payload?.value) === 'object')) {
```

Comment:
is this condition really needed? aren't we validating in the constructor?

Reply:
Unfortunately, I think this is required, which I agree isn't ideal. Constructor also permits strings, which in almost all cases is valid, except this one.

Reply:
oh i see, perhaps let's add a comment since it's not entirely intuitive why it's needed at first blush